### PR TITLE
DropDownBox: Use correct popup maxHeight if popup is already rendered (T874949)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/drop_down_box.js [hunk 1]
```diff
@@ -272,7 +272,9 @@ const DropDownBox = DropDownEditor.inherit({
             },
             onKeyboardHandled: opts => this.option('focusStateEnabled') && this._popupElementTabHandler(opts),
             maxHeight: function() {
-                return getElementMaxHeightByWindow(this.$element());
+                const popupLocation = (this._popup && this._popup._resultPosition) ? this._popup._resultPosition.v.location : undefined;
```
Comment (DokaRus) [inline js/ui/drop_down_box.js]:
```suggestion
                const popupLocation = this?._popup?._resultPosition ? this._popup._resultPosition.v.location : undefined;
```

File: js/ui/drop_down_box.js [hunk 2]
```diff
@@ -272,7 +272,9 @@ const DropDownBox = DropDownEditor.inherit({
             },
             onKeyboardHandled: opts => this.option('focusStateEnabled') && this._popupElementTabHandler(opts),
             maxHeight: function() {
-                return getElementMaxHeightByWindow(this.$element());
+                const popupLocation = this?._popup?._resultPosition ? this._popup._resultPosition.v.location : undefined;
```
Comment (San4es) [inline js/ui/drop_down_box.js]: Can we save a position on the `onPositioned` callback?
```js
onPositioned: function(e) {
    this._popupPosition = e.position;
}.bind(this),
maxHeight: function() {
    const popupLocation = this._popupPosition?.v.location;
    return getElementMaxHeightByWindow(this.$element(), popupLocation);
}.bind(this)
```

Comment (alexander-kotov-dx) [reply js/ui/drop_down_box.js]: No, it overwrites dropDownEditor handler. But we can extend _popupPositionedHandler

File: js/ui/drop_down_box.js [hunk 3]
```diff
@@ -272,7 +282,9 @@ const DropDownBox = DropDownEditor.inherit({
             },
             onKeyboardHandled: opts => this.option('focusStateEnabled') && this._popupElementTabHandler(opts),
             maxHeight: function() {
-                return getElementMaxHeightByWindow(this.$element());
+                const popupLocation = this?._popupPosition ? this._popupPosition.v.location : undefined;
```
Comment (San4es) [inline js/ui/drop_down_box.js]:
```suggestion
                const popupLocation = this._popupPosition?.v.location;
```

?

File: js/ui/drop_down_box.js [hunk 4]
```diff
@@ -248,6 +248,16 @@ const DropDownBox = DropDownEditor.inherit({
         return realDevice.deviceType === 'desktop' && this._canShowVirtualKeyboard() && this._isNestedElementActive();
     },
 
+    _popupHiddenHandler: function() {
+        this.callBase();
+        this._resultPosition = undefined;
```
Comment (San4es) [inline js/ui/drop_down_box.js]: Why should we do this?

**Review by DokaRus — DISMISSED** [review]

**Review by San4es — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**